### PR TITLE
Do not catch exception when there is error in put/delete for old table

### PR DESCRIPTION
**Pull request by mikeyang923**
Throw exception when there is conflict error in put for old table

### CODE REVIEW

File: ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java [hunk 1]
```diff
@@ -337,8 +337,8 @@ public CompletableFuture<PutResult> put(NamedBlobRecord record, NamedBlobState s
           PutResult putResult = null, putResultV2 = null;
           try {
             putResult = run_put(record, accountId, containerId, connection);
-          } catch (Exception eOld) {
-            if (!config.dbRelyOnNewTable) {
+          } catch (RestServiceException eOld) {
+            if (!config.dbRelyOnNewTable || eOld.getErrorCode() == RestServiceErrorCode.Conflict) {
```
Comment (snalli) [inline ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java:341]: whenever the boolean grows long, make a helper. Its likely it'll grow.
i don't know what this expression means but a readable helper would instantly convey the meaning.
I can think of the following:
```
void onConflictError(eOld) throws RestServiceException {
  // your logic to throw here
}
```

Comment (snalli) [reply ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java:341]: And please make a new catch block for this exception.

Comment (snalli) [reply ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java:341]:
```
catch (NewException) {
 onNewException()
}
```
something like this

**Comment by justinlin-linkedin** [comment]
what is the best way going forward? If we want to compare content fetched from both old and new tables, maybe we should always stop the transaction as long as one of the table throws exception, regardless what exception they are throwing.

**Comment by mikeyang923** [comment]
> what is the best way going forward? If we want to compare content fetched from both old and new tables, maybe we should always stop the transaction as long as one of the table throws exception, regardless what exception they are throwing.

Yeah, I agree, I just updated, can you take another look?

File: ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java (hunk 1 above)
Comment (mikeyang923) [reply ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java]: This is outdated, I updated per Justin's comment in 
2e39a00. Can you take another look?

**Review by justinlin-linkedin — APPROVED** [review]
LGTM

**Review by panchow — APPROVED** [review]

**Review by snalli — APPROVED** [review]
if this is the only change, then lgtm

**Comment by snalli** [comment]
actually mike, you may wanna change the title of the PR now since we are not throwing any exceptions

**Review by snalli — CHANGES_REQUESTED** [review]
actually mike, you may wanna change the title of the PR now since we are not throwing any exceptions

**Comment by mikeyang923** [comment]
> actually mike, you may wanna change the title of the PR now since we are not throwing any exceptions

Good point, just updated the PR title. Can you take another look?

**Pull request merged**